core/src/Cuda/Kokkos_Cuda_Instance.hpp

```diff
@@ -102,7 +105,7 @@ class CudaInternal {
  public:
   using size_type = Cuda::size_type;
 
-  inline static int m_cudaDev = -1;
+  int m_cudaDev = -1;
 
   // Device Properties
```

Comment:
We still assume that all the devices used have the same properties. Dropping that assumption would require the following variables also to be per device (and some more changes when querying the functor properties for example).